.circleci/config.yml

```diff
@@ -53,45 +66,76 @@ jobs:
           root: .
           paths:
             - .
-  build:
+  build-for-testing:
     macos:
       xcode: *xcode_version
     resource_class: *resource_class
     steps:
       - attach_workspace:
           at: .
-      - install-tools # chaching currently fails, needs some further tweaks
       - run:
-          name: Build
-          command: cd src/xcode && bundle exec fastlane build
+          name: Build for testing
+          command: |
+            cd src/xcode && \
+            xcodebuild \
+              -workspace "ENA.xcworkspace" \
+              -scheme ENA \
+              -destination "platform=iOS Simulator,OS=14.4,name=iPhone 11" \
+              -testPlan AllTests \
+              -only-test-configuration "DE" \
+              -derivedDataPath "./DerivedData" \
+              -jobs `sysctl -n hw.ncpu` \
+              -parallelizeTargets \
+              -quiet \
+              build-for-testing
+      - persist_to_workspace:
+          root: .
+          paths:
+            - "src/xcode/DerivedData/**/*"
+  swiftlint:
+    macos:
+      xcode: *xcode_version
+    resource_class: medium
+    steps:
+      - attach_workspace:
+          at: .
+      - restore_cache:
+          key: *cache_key
+      - restore_cache:
+          key: *tools_cache_key
+      - run:
+          name: Swiftlint
+          command: cd src/xcode && bundle exec fastlane lint
+      - store_artifacts:
+          path: src/xcode/swiftlint.result.json
   test:
     macos:
       xcode: *xcode_version
     resource_class: *resource_class
+    parameters:
+      testplan:
+        default: "AllTests"
+        type: string
     steps:
       - attach_workspace:
           at: .
-      - install-tools # chaching currently fails, needs some further tweaks
+      - restore_cache:
+          key: *tools_cache_key
       - run:
           name: Run Fastlane tests
           command: cd src/xcode && bundle exec fastlane test
-          no_output_timeout: 25m
+          no_output_timeout: 5m
```

Comment:
I changed the output style in Fastlane to "spam" a little more. We should now see quicker when the CI is really stuck or just repeating tests due to Apple's wacky test framework.